Digitall.Warehouse.Application/Categories/Commands/CreateCategoryCommandHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Digitall.Warehouse.Application.Abstractions.Messaging;
+using Digitall.Warehouse.Application.Abstractions.Persistence;
+using Digitall.Warehouse.Domain.Entities.Products;
+using Digitall.Warehouse.Domain.Shared;
+
+namespace Digitall.Warehouse.Application.Categories.Commands
+{
+    public class CreateCategoryCommandHandler : ICommandHandler<CreateCategoryCommand>
+    {
+        private readonly ICategoryRepository _categoryRepository;
+        private readonly IUnitOfWork _unitOfWork;
+
+        public CreateCategoryCommandHandler(
+            ICategoryRepository categoryRepository,
+            IUnitOfWork unitOfWork)
+        {
+            _categoryRepository = categoryRepository;
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task<Result> Handle(
+            CreateCategoryCommand request,
+            CancellationToken cancellationToken)
+        {
+            var category = new Category()
+            {
+                Name = request.Name,
+            };
+
+            await _categoryRepository.AddAsync(category, cancellationToken);
+            await _unitOfWork.SaveChangesAsync();
```

Comment:
You don't need to inject every repository you want to save something .
You need only unitofwork and the code should looks like this

 await _unitOfWork.Categories.AddAsync(category, cancellationToken);
 await _unitOfWork.SaveChangesAsync();

Reply:
Yep, could be done that way too. Will do this code change as new PR right after this one is merged.

Reply:
I've done it you could check #4 commit: [9422b5ae8dde7df3038bdf6e8051ec64c7f26c0d](https://github.com/tpilitis/Digitall.Warehouse/pull/4/commits/f8bb2cb608688aab79f22c7789ec5dc313d29064)